qtawesome/iconic_font.py

```diff
@@ -206,17 +236,70 @@ def _parse_options(self, specific_options, general_options, name):
+        disabled = options.get('disabled', char)
+        on_active = options.get('on_active', active)
+        on_selected = options.get('on_selected', selected)
+        on_disabled = options.get('on_disabled', disabled)
+        off_active = options.get('off_active', active)
+        off_selected = options.get('off_selected', selected)
+        off_disabled = options.get('off_disabled', disabled)
+
+        icon_dict = {'char': char,
+                     'on': on,
+                     'off': off,
+                     'active': active,
+                     'selected': selected,
+                     'disabled': disabled,
+                     'on_active': on_active,
+                     'on_selected': on_selected,
+                     'on_disabled': on_disabled,
+                     'off_active': off_active,
+                     'off_selected': off_selected,
+                     'off_disabled': off_disabled,
+                     }
+        names = [icon_dict.get(kw, name) for kw in icon_kw]
         prefix, chars = self._get_prefix_chars(names)
         options.update(dict(zip(*(icon_kw, chars))))
         options.update({'prefix': prefix})
 
-        # Handle colors for states
-        color_kw = ['color_active', 'color_selected']
-        colors = [options.get(kw, options['color']) for kw in color_kw]
-        options.update(dict(zip(*(color_kw, colors))))
+        # Handle colors for modes (Active, Disabled, Selected, Normal)
+        # and states (On, Off)
+        color = options.get('color')
+        color_on = options.get('color_on', color)
+        color_off = options.get('color_off', color)
+        color_active = options.get('color_active', color_on)
+        color_selected = options.get('color_selected', color_active)
+        color_disabled = options.get('color_disabled')
+        color_on_active = options.get('color_on_active', color_active)
+        color_on_selected = options.get('color_on_selected', color_selected)
+        color_on_disabled = options.get('color_on_disabled', color_disabled)
+        color_off_active = options.get('color_off_active', color_active)
+        color_off_selected = options.get('color_off_selected', color_selected)
+        color_off_disabled = options.get('color_off_disabled', color_disabled)
+
+        color_dict = {'color_active': color_active,
+                      'color_selected': color_selected,
+                      'color_disabled': color_disabled,
+                      'color_on': color_on,
+                      'color_on_active': color_on_active,
+                      'color_on_selected': color_on_selected,
+                      'color_on_disabled': color_on_disabled,
+                      'color_off': color_off,
+                      'color_off_active': color_off_active,
+                      'color_off_selected': color_off_selected,
+                      'color_off_disabled': color_off_disabled,
+                      }
+        options.update(color_dict)
```

Comment:
`setdefault` defines the item if it doesn't exist, so instead of getting the options, building a dict and updating the options, you directly set them if needed. (Or I didn't understand what the code does)